module/zfs/metaslab.c

```diff
@@ -1004,16 +1008,23 @@ metaslab_group_initialized(metaslab_group_t *mg)
 uint64_t
 metaslab_group_get_space(metaslab_group_t *mg)
 {
-	return ((1ULL << mg->mg_vd->vdev_ms_shift) * mg->mg_vd->vdev_ms_count);
+	/*
+	 * Note that the number of nodes in mg_metaslab_tree may be one less
+	 * than vdev_ms_count, due to the embedded log metaslab.
+	 */
+	mutex_enter(&mg->mg_lock);
+	uint64_t space = (1ULL << mg->mg_vd->vdev_ms_shift) *
+	    avl_numnodes(&mg->mg_metaslab_tree);
+	mutex_exit(&mg->mg_lock);
+	return (space);
```

Comment:
Consider:
```
    mutex_enter(@mg->mg_lock);
    uint64_t ms_count = avl_numnodes(&mg->mg_metaslab_tree);
    mutex_exit(@mg->mg_lock);
    return ((1ULL << mg->mg_vd->vdev_ms_shift) * ms_count);
```
I think this is slightly clearer (particularly wrt the comment) and holds the mg_lock a slightly shorter period of time.